cli/src/main/java/com/liferay/blade/cli/command/ConvertCommand.java

```diff
@@ -689,4 +810,117 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) th
 		return pluginDir[0];
 	}
 
+	private File _getPluginsSdkDir(ConvertArgs convertArgs, File projectDir, Properties gradleProperties) {
+		File pluginsSdkDir = convertArgs.getSource();
+
+		if (pluginsSdkDir == null) {
+			if (gradleProperties != null) {
+				String pluginsSdkDirValue = gradleProperties.getProperty(
+					WorkspaceConstants.DEFAULT_PLUGINS_SDK_DIR_PROPERTY);
+
+				if (pluginsSdkDirValue != null) {
+					pluginsSdkDir = new File(projectDir, pluginsSdkDirValue);
+				}
+			}
+
+			if (pluginsSdkDir == null) {
+				pluginsSdkDir = new File(projectDir, WorkspaceConstants.DEFAULT_PLUGINS_SDK_DIR);
+			}
+		}
+
+		return pluginsSdkDir;
+	}
+
+	private boolean _isValidSDKDir(File pluginsSdkDir) {
+		File buildProperties = new File(pluginsSdkDir, "build.properties");
+		File portletsBuildXml = new File(pluginsSdkDir, "portlets/build.xml");
+		File hooksBuildXml = new File(pluginsSdkDir, "hooks/build.xml");
+
+		if (buildProperties.exists() && portletsBuildXml.exists() && hooksBuildXml.exists()) {
+			return true;
+		}
+
+		return false;
+	}
+
+	private Properties _loadProperties(InputStream inputStream) throws IOException {
+		Properties properties = new Properties();
+
+		properties.load(inputStream);
+
+		inputStream.close();
+
+		return properties;
+	}
+
+	private void _warn(String message) {
+		BladeCLI bladeCLI = getBladeCLI();
+
+		bladeCLI.out("WARNING: " + message);
+	}
+
+	private static class GAV {
+
```

Comment:
I don't think it is good Java naming conventions to have a fully upper-case class, perhaps we could rename it GroupArtifactVersion, or something like that?